assets/src/blocks/PostsListTest/template.js

```diff
@@ -0,0 +1,95 @@
+export const POSTS_LIST_BLOCK_NAME = 'planet4-blocks/posts-list';
+export const POSTS_LISTS_LAYOUT_TYPES = [
+  {label: 'List', value: 'default', columnCount: 3},
+  {label: 'Grid', value: 'grid', columnCount: 4},
+  {label: 'Carousel', value: 'flex', columnCount: 8},
+];
+
+const {__} = wp.i18n;
+
+const newsPageLink = window.p4_vars.news_page_link;
+
+const seeAllLink = ['core/navigation-link', {...!newsPageLink ? {className: 'd-none'} : {
+  url: newsPageLink,
+  label: __('See all stories', 'planet4-blocks-backend'),
+  className: 'see-all-link',
+}}];
+
+const template = () => ([
+  ['core/query', {
+    className: 'posts-list p4-query-loop is-custom-layout-list',
+    query: {
+      perPage: 3,
+      pages: 0,
+      offset: 0,
+      postType: 'post',
+      order: 'desc',
+      orderBy: 'date',
+      author: '',
+      search: '',
+      exclude: [],
+      sticky: '',
+      inherit: false,
+      postIn: [],
+      hasPassword: false,
+    },
+    layout: {
+      type: 'default',
+      columnCount: 3,
+    },
+  }, [
+    ['core/group', {layout: {type: 'flex', justifyContent: 'space-between'}}, [
+      ['core/heading', {lock: {move: true}, content: __('Related Posts', 'planet4-blocks-backend')}],
+      seeAllLink,
+    ]],
+    ['core/paragraph', {
+      lock: {move: true},
+      placeholder: __('Enter description', 'planet4-blocks-backend'),
+      style: {
+        spacing: {
+          margin: {
+            top: '24px',
+            bottom: '36px',
+          },
+        },
+      },
+    }],
+    ['core/query-no-results', {}, [
+      ['core/paragraph', {content: __('No posts found. (This default text can be edited)', 'planet4-blocks-backend')}],
+    ]],
+    ['core/post-template', {lock: {move: true, remove: true}}, [
+      ['core/columns', {}, [
```

Comment:
Thinking loud.. do we really need to use `columns` here? Wouldn't be better if we use a `group`?